cmd/fetchd/cmd/genasiupgrade.go

```diff
@@ -41,23 +35,54 @@ const (
 	ValAddressPrefix  = "valoper"
 	ConsAddressPrefix = "valcons"
 
-	NewBaseDenom   = "asi"
-	NewDenom       = "aasi"
-	NewAddrPrefix  = "asi"
-	NewChainId     = "asi-1"
-	NewDescription = "ASI Token" // TODO(JS): change this, potentially
-
-	OldDenom      = "afet"
+	NewAddrPrefix = "asi"
 	OldAddrPrefix = "fetch"
 )
 
+var networkInfos = map[string]NetworkConfig{
+	"mainnet": {
+		NewChainID:     "asi-1",
+		NewDescription: "ASI token", // TODO(JS): confirm this
+		DenomInfo: DenomInfo{
+			NewBaseDenom: "asi",
+			NewDenom:     "aasi",
+			OldDenom:     "afet",
+		},
+		IbcTargetAddr:            "fetch1rhrlzsx9z865dqen8t4v47r99dw6y4va4uph0x", // TODO(JS): amend this
+		ReconciliationTargetAddr: "fetch1rhrlzsx9z865dqen8t4v47r99dw6y4va4uph0x", // TODO(JS): amend this
+		Contracts: Contracts{
+			TokenBridge: TokenBridge{
+				Addr:     "fetch1qxxlalvsdjd07p07y3rc5fu6ll8k4tmetpha8n",
+				NewAdmin: "fetch15p3rl5aavw9rtu86tna5lgxfkz67zzr6ed4yhw",
+			},
+		},
+	},
+	"testnet": {
+		NewChainID:     "dorado-1",       // TODO(JS): confirm this
+		NewDescription: "Test ASI token", // TODO(JS): confirm this
+		DenomInfo: DenomInfo{
+			NewBaseDenom: "testasi",
+			NewDenom:     "atestasi",
+			OldDenom:     "atestfet",
+		},
+		IbcTargetAddr:            "fetch1rhrlzsx9z865dqen8t4v47r99dw6y4va4uph0x", // TODO(JS): amend this
+		ReconciliationTargetAddr: "",                                             // Not applicable to testnet
+		Contracts: Contracts{
+			TokenBridge: TokenBridge{
+				Addr:     "fetch1qxxlalvsdjd07p07y3rc5fu6ll8k4tmetpha8n", // TODO(JS): amend this
+				NewAdmin: "fetch15p3rl5aavw9rtu86tna5lgxfkz67zzr6ed4yhw", // TODO(JS): amend this
+			},
+		},
+	},
+}
+
 //go:embed reconciliation_data.csv
 var reconciliationData []byte
 
 // ASIGenesisUpgradeCmd returns replace-genesis-values cobra Command.
 func ASIGenesisUpgradeCmd(defaultNodeHome string) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "asi-genesis-upgrade",
+		Use:   "asi-genesis-upgrade [testnet|mainnet]",
```

Comment:
You can drop the `[testnet|mainnet]` paremeter (the decission wil be made automatically based on the `chain-id` value in the input genesis.json file.